Exam/CC1/src/exercice4/Reservation.java

```diff
@@ -0,0 +1,46 @@
+package exercice4;
+
+import java.util.Date;
+
+public class Reservation {
+
+	
+	public String proprietaire;
+	public String chambre;
+	public String nom;
+	public Date heure;
+	
+	public Reservation(String proprietaire, String chambre, String nom, Date heure) {
+		this.proprietaire = proprietaire;
+		this.chambre = chambre;
+		this.nom = nom;
+		this.heure = heure;
+	}
+
+	public void update(Reservation updt) {
+		if (this.equals(updt)) {
+			proprietaire = updt.proprietaire;
+			nom = updt.nom;
+		}
+		else {
+			
+			return;
+		}
+	}
+	
+	@Override
+	public boolean equals(Object temp) {
+		if (!(temp instanceof Reservation)) {
+			return super.equals(temp);
+		}
+		else {
+			Reservation b = (Reservation) temp;
+			return chambre.equals(b.chambre) && heure.equals(b.heure);
```

Comment:
So a `Reservation` object is uniquely identified by `(chambre,heure)`...
Why not... But an id would have been simpler?